File: ax-request-hub/scripts/seed-governance-docs.py
"""
거버넌스 문서 DB 메타데이터 시드 스크립트
- 기존 GovernanceDoc 전체 삭제 후 재삽입
- fileName = docs/ 기준 상대경로 (governance/full/... 형태)
"""
import sqlite3, os, sys
from datetime import datetime

DB_PATH = os.path.join(os.path.dirname(__file__), '..', 'prisma', 'dev.db')

def cuid_like(n):
    import random, string
    chars = string.ascii_lowercase + string.digits
    return 'c' + ''.join(random.choices(chars, k=24))

NOW = datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
APPROVED = '2026-01-01 00:00:00'

DOCS = [
    # (docId, fileName, type, level, title, version, securityLevel, status, description)
    # ── L1 규정 ──────────────────────────────────────────────────
    ('AX-REG-2026-001',
     'governance/full/AX-REGULATION-2026-001_AI운영규정_v4.0.md',
     '규정', 'L1',
     'AI 운영 규정 v4.0',
     'v4.0', 'G2', 'active',
     'AX Hub 전체 AI 운영의 최상위 규정. 거버넌스 원칙·범위·책임체계 정의'),
    ('AX-COM-2026-001',
     'governance/full/AX-COMMITTEE-2026-001_AI운영위원회규정_v1.2.md',
     '규정', 'L1',
     'AI 운영위원회 규정 v1.2',
     'v1.2', 'G2', 'active',
     'AI 운영위원회 구성·권한·의사결정 절차 정의'),
    # ── L2 지침·운영방안 ─────────────────────────────────────────
    ('AX-POL-2026-001',
     'governance/full/AX-POLICY-2026-001_AI운영지침_v6.5.md',
     '지침', 'L2',
     'AI 운영 지침 v6.5',
     'v6.5', 'G2', 'active',
     'AI 운영 정책의 세부 지침. 에이전트 운영·데이터 거버넌스·보안 기준 포함'),
    ('AX-STD-2026-001',
     'governance/full/AX-STANDARD-2026-001_AI운영기준_v1.1.md',
     '운영방안', 'L2',
     'AI 운영 기준 v1.1',
     'v1.1', 'G2', 'active',
     'AI 시스템 운영 표준·KPI·SLA 기준 정의'),
    # ── L3 가이드·매뉴얼 ─────────────────────────────────────────
    ('AX-MAN-2026-001',
     'governance/full/AX-MANUAL-2026-001_사용가이드라인.md',
     '매뉴얼', 'L3',
     'AI 활용 사용자 가이드라인',
     'v1.0', 'G1', 'active',
     '임직원 대상 AI 도구 사용 가이드라인'),
    ('AX-OPS-2026-001',
     'governance/operations/AI에이전트등록_운영가이드.md',
     '가이드라인', 'L3',
     'AI 에이전트 등록 운영가이드',
     'v1.0', 'G2', 'active',
     'AX Hub 에이전트 등록·라이프사이클 관리 절차'),
    ('AX-OPS-2026-002',
     'governance/operations/AX_AI도구계정_관리운영가이드.md',
     '가이드라인', 'L3',
     'AI 도구 계정 관리 운영가이드',
     'v1.0', 'G2', 'active',
     'AI 도구 계정 발급·권한관리·폐기 절차'),
    ('AX-OPS-2026-003',
     'governance/operations/생성형AI모델배분_운영가이드.md',
     '가이드라인', 'L3',
     '생성형 AI 모델 배분 운영가이드',
     'v1.0', 'G2', 'active',
     '생성형 AI 모델 선택·토큰 배분·비용 관리 절차'),
    ('AX-STD-2026-002',
     'governance/operations/AX_AI리터러시레벨_운영기준.md',
     '운영방안', 'L3',
     'AI 리터러시 레벨 운영기준',
     'v1.0', 'G2', 'active',
     '임직원 AI 역량 레벨 체계·평가 기준 정의'),
    ('AX-STD-2026-003',
     'governance/operations/AX_스킬라이브러리_운영기준.md',
     '운영방안', 'L3',
     'AX 스킬 라이브러리 운영기준',
     'v1.0', 'G2', 'active',
     'AI 스킬 등록·버전관리·배포 운영 기준'),
    ('AX-STD-2026-004',
     'governance/operations/AX_토큰정책_운영기준.md',
     '운영방안', 'L3',
     'AX 토큰 정책 운영기준',
     'v1.0', 'G2', 'active',
     'LLM 토큰 사용 한도·비용 배분·모니터링 정책'),
    ('AX-DEV-2026-001',
     'governance/operations/AX_AI개발플로우_추가조항.md',
     '개발표준', 'L3',
     'AI 개발 플로우 추가 조항',
     'v1.0', 'G2', 'active',
     'AI 과제 개발 절차의 추가 조항·예외처리 기준'),
    ('AX-DEV-2026-002',
     'governance/operations/AX_개발표준_전사AI과제용.md',
     '개발표준', 'L3',
     '전사 AI 과제 개발 표준',
     'v1.0', 'G2', 'active',
     '전사 AI 과제에 적용되는 개발 코드·문서·테스트 표준'),
]

conn = sqlite3.connect(DB_PATH)
cur = conn.cursor()

# 기존 레코드 전체 삭제
cur.execute('DELETE FROM GovernanceDoc')
print(f'기존 레코드 삭제 완료')

# 새 레코드 삽입
inserted = 0
for (doc_id, file_name, dtype, level, title, version, sec_level, status, desc) in DOCS:
    record_id = cuid_like(0)
    cur.execute('''
        INSERT INTO GovernanceDoc (
            id, docId, fileName, type, level, title, version,
            author, approvedBy, approvedAt, securityLevel, status,
            description, relatedDocs, createdAt, updatedAt
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
    ''', (
        record_id, doc_id, file_name, dtype, level, title, version,
        'AX팀', 'AX 운영위원회', APPROVED, sec_level, status,
        desc, '[]', NOW, NOW
    ))
    print(f'  [{doc_id}] {title}')
    inserted += 1

conn.commit()
conn.close()
print(f'\n총 {inserted}개 문서 메타데이터 등록 완료')

File: ax-request-hub/scripts/seed-governance-docs.mjs
/**
 * 거버넌스 문서 시드 스크립트 (직접 실행용)
 * node scripts/seed-governance-docs.mjs
 */
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()

const DEPRECATED_IDS = [
  'AX-REG-2026-001', 'AX-REG-2026-002', 'AX-OPS-2026-001',
  'AX-GUI-2026-001', 'AX-GUI-2026-002', 'AX-MAN-2026-001', 'AX-DEV-2026-001',
]

const SEED_DOCS = [
  // L1 규정
  { docId: 'AI-REG-2026-001', fileName: '[규정]_AI운영규정_ai-reg-2026-001.md', type: '규정', level: 'L1', title: 'AI 운영 규정', version: 'v1.0', author: 'AX팀', approvedBy: 'AX/PI센터장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '삼성자산운용 전사 AI 운영의 기본 규정', relatedDocs: ['AI-REG-2026-002', 'AI-REG-2026-003', 'AI-GUI-2026-001'] },
  { docId: 'AI-REG-2026-002', fileName: '[규정]_AI위원회규정_ai-reg-2026-002.md', type: '규정', level: 'L1', title: 'AI 위원회 규정', version: 'v1.0', author: 'AX팀', approvedBy: 'AX/PI센터장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: 'AI 위원회 구성·운영·의결 절차 규정', relatedDocs: ['AI-REG-2026-001', 'AI-REG-2026-003'] },
  { docId: 'AI-REG-2026-003', fileName: '[규정]_AI위험관리규정_ai-reg-2026-003.md', type: '규정', level: 'L1', title: 'AI 위험관리 규정', version: 'v1.0', author: 'AX팀', approvedBy: 'AX/PI센터장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: 'AI 위험 유형(R-01~R-11)·고영향 AI·FRIA 법적 근거 규정', relatedDocs: ['AI-REG-2026-001', 'AI-GUI-2026-002'] },
  // L2 지침
  { docId: 'AI-GUI-2026-001', fileName: '[지침]_AI운영지침_ai-gui-2026-001.md', type: '지침', level: 'L2', title: 'AI 운영 지침', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '과제 신청·승인·Gate·에이전트 등록(제16조) 세부 절차', relatedDocs: ['AI-REG-2026-001', 'AI-GUI-2026-002', 'AI-STD-2026-001'] },
  { docId: 'AI-GUI-2026-002', fileName: '[지침]_AI위험관리지침_ai-gui-2026-002.md', type: '지침', level: 'L2', title: 'AI 위험관리 지침', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '위험 매트릭스·FRIA·투명성 표시 이행 기준(제12조) 세부 절차', relatedDocs: ['AI-REG-2026-003', 'AI-GUI-2026-001'] },
  // L3 운영기준
  { docId: 'AI-STD-2026-001', fileName: '[가이드라인]_AI운영기준_ax-std-2026-001.md', type: '운영기준', level: 'L3', title: 'AI 운영기준 (별표)', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '승인 점수·기밀등급·Gate 기준 수치 SSOT', relatedDocs: ['AI-GUI-2026-001'] },
  { docId: 'AI-STD-2026-002', fileName: '[가이드라인]_AI리터러시레벨운영기준_ax-std-2026-002.md', type: '운영기준', level: 'L3', title: 'AI 리터러시 레벨 운영기준', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: 'L0~L4 리터러시 레벨 정의·취득 기준', relatedDocs: ['AI-GUI-2026-001'] },
  { docId: 'AI-STD-2026-003', fileName: '[가이드라인]_AX스킬라이브러리운영기준_ax-std-2026-003.md', type: '운영기준', level: 'L3', title: 'AX 스킬 라이브러리 운영기준', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '스킬 6단계 표준(탐색→배포) 및 카탈로그 관리 기준', relatedDocs: ['AI-GUI-2026-001'] },
  { docId: 'AI-STD-2026-004', fileName: '[가이드라인]_AX토큰정책운영기준_ax-std-2026-004.md', type: '운영기준', level: 'L3', title: 'AX 토큰 정책 운영기준', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '월별 토큰 한도·경고 임계치·초과 정책', relatedDocs: ['AI-GUI-2026-001'] },
  // L3 가이드
  { docId: 'AI-OPS-2026-001', fileName: '[가이드라인]_AI에이전트등록운영가이드_ax-ops-2026-001.md', type: '가이드', level: 'L3', title: 'AI 에이전트 등록 운영가이드', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '유형 3·4 에이전트 등록 신청·검토·Agent ID 부여 절차', relatedDocs: ['AI-GUI-2026-001', 'AI-REG-2026-003'] },
  { docId: 'AI-OPS-2026-002', fileName: '[가이드라인]_AI도구계정관리운영가이드_ax-ops-2026-002.md', type: '가이드', level: 'L3', title: 'AI 도구 계정관리 운영가이드', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: 'ChatGPT·Copilot·Claude 등 AI 도구 계정 배분·회수·관리 기준', relatedDocs: ['AI-GUI-2026-001'] },
  { docId: 'AI-OPS-2026-003', fileName: '[가이드라인]_생성형AI모델배분운영가이드_ax-ops-2026-003.md', type: '가이드', level: 'L3', title: '생성형 AI 모델 배분 운영가이드', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '업무 유형별 모델 배분 기준 및 순량제·정액제 운영 원칙', relatedDocs: ['AI-GUI-2026-001', 'AI-STD-2026-004'] },
  // L3 개발표준
  { docId: 'AI-DEV-2026-001', fileName: '[가이드라인]_AI개발플로우추가조항_ax-dev-2026-001.md', type: '개발표준', level: 'L3', title: 'AI 개발플로우 추가 조항', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: 'AI 과제 개발 프로세스 보완 조항 (Gate 2 기술표준 자가점검 포함)', relatedDocs: ['AI-GUI-2026-001', 'AI-DEV-2026-002'] },
  { docId: 'AI-DEV-2026-002', fileName: '[가이드라인]_전사AI과제개발표준_ax-dev-2026-002.md', type: '개발표준', level: 'L3', title: '전사 AI 과제 개발 표준', version: 'v1.0', author: 'AX팀 · IT업무개발팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G2', status: 'active', description: '전사 AI 과제 개발 시 준수해야 할 기술 표준 및 체크리스트', relatedDocs: ['AI-DEV-2026-001'] },
  // L3 매뉴얼
  { docId: 'AI-MAN-2026-001', fileName: '[매뉴얼]_AX허브사용안내_ax-man-2026-001.md', type: '매뉴얼', level: 'L3', title: 'AX Hub 사용 안내 매뉴얼', version: 'v1.0', author: 'AX팀', approvedBy: 'AX팀장', approvedAt: '2026-09-01', securityLevel: 'G1', status: 'active', description: '전 직원 대상 AX Request Hub 사용 방법 안내', relatedDocs: ['AI-GUI-2026-001'] },
  // L4 기술문서
  { docId: 'AX-TEC-2026-001', fileName: 'AX_거버넌스_문서체계.md', type: '기술문서', level: 'L4', title: 'AX 거버넌스 문서 체계', version: 'v1.0', author: 'AX팀', approvedBy: '', approvedAt: null, securityLevel: 'G1', status: 'active', description: 'AX Hub 거버넌스 문서 분류 체계 및 관리 원칙', relatedDocs: [] },
  { docId: 'AX-TEC-2026-003', fileName: 'architecture.md', type: '기술문서', level: 'L4', title: 'AX Hub 시스템 아키텍처', version: 'v1.0', author: 'IT업무개발팀', approvedBy: '', approvedAt: null, securityLevel: 'G2', status: 'active', description: 'AX Hub Next.js 앱 아키텍처 및 모듈 구조 설명', relatedDocs: [] },
  { docId: 'AX-TEC-2026-004', fileName: 'registry-lifecycle-design.md', type: '기술문서', level: 'L4', title: 'ETF 에이전트 레지스트리 라이프사이클 설계', version: 'v1.0', author: 'IT업무개발팀', approvedBy: '', approvedAt: null, securityLevel: 'G2', status: 'active', description: 'ETF 앙상블 에이전트 Gate 라이프사이클 설계 문서', relatedDocs: [] },
]

async function main() {
  console.log('거버넌스 문서 시드 시작...')

  // 1. 구버전 deprecated 처리
  let deprecated = 0
  for (const oldId of DEPRECATED_IDS) {
    const result = await prisma.governanceDoc.updateMany({
      where: { docId: oldId, status: { not: 'deprecated' } },
      data: { status: 'deprecated' },
    })
    deprecated += result.count
    if (result.count > 0) console.log(`  deprecated: ${oldId}`)
  }

  // 2. 신규 문서 upsert
  const seeded = []
  for (const d of SEED_DOCS) {
    const doc = await prisma.governanceDoc.upsert({
      where: { docId: d.docId },
      create: {
        docId: d.docId, fileName: d.fileName, type: d.type, level: d.level,
        title: d.title, version: d.version, author: d.author, approvedBy: d.approvedBy,
        securityLevel: d.securityLevel, status: d.status, description: d.description,
        relatedDocs: JSON.stringify(d.relatedDocs ?? []),
        approvedAt: d.approvedAt ? new Date(d.approvedAt) : null,
      },
      update: {
        fileName: d.fileName, type: d.type, level: d.level,
        title: d.title, version: d.version, author: d.author, approvedBy: d.approvedBy,
        securityLevel: d.securityLevel, status: d.status, description: d.description,
        relatedDocs: JSON.stringify(d.relatedDocs ?? []),
        approvedAt: d.approvedAt ? new Date(d.approvedAt) : null,
      },
    })
    seeded.push(doc.docId)
    console.log(`  upserted: ${doc.docId} — ${d.title}`)
  }

  console.log(`\n완료: deprecated ${deprecated}개, upserted ${seeded.length}개`)
}

main()
  .catch(e => { console.error(e); process.exit(1) })
  .finally(() => prisma.$disconnect())
